bitcoin/src/address.rs

```diff
@@ -565,22 +632,75 @@ impl<'a> fmt::Display for AddressEncoding<'a> {
 /// * [BIP341 - Taproot: SegWit version 1 spending rules](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki)
 /// * [BIP350 - Bech32m format for v1+ witness addresses](https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki)
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct Address {
+pub struct Address<V = NetworkChecked>
+where
+    V: NetworkValidation,
+{
     /// The type of the address.
     pub payload: Payload,
     /// The network on which this address is usable.
     pub network: Network,
+    /// Marker of the status of network validation.
+    _validation: PhantomData<V>,
 }
+
+#[cfg(feature = "serde")]
+crate::serde_utils::serde_string_impl!(Address<NetworkUnchecked>, "a Bitcoin address");
+
+// Macro serde_string_impl creates Serialize and Deserialize for Address<NetworkUnchecked>
+// Here we add Serialize for Address<NetworkChecked>. There is no Deserialize for it.
 #[cfg(feature = "serde")]
-crate::serde_utils::serde_string_impl!(Address, "a Bitcoin address");
+impl serde::Serialize for Address {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.collect_str(&self)
+    }
+}
 
+/// Methods on [`Address`] that can be called on both `Address<NetworkChecked>` and
+/// `Address<NetworkUnchecked>`.
```

Comment:
Does this doc comment have any effect?

Reply:
This, and other two on the other `impl Addr<…>`, were originally plain comments but I turned them into doc as a hint to readers of documentation that implementation of methods and functions are spread across multiple impls.

Would you prefer another way? Not having them as doc comments?

Reply:
I was just curious, if it has no effect it doesn't matter which kind of comment it is.

Reply:
It gets rendered in the documentation at the beginning of the impl.